### PR TITLE
Only increment invocation usage after inserting a new invocation row

### PR DESCRIPTION
**Pull request by bduffany**
This should provide a stronger guarantee that the usage count does not exceed the number of rows inserted into the DB (modulo the possibility of deleting invocations via the UI).

---

**Version bump**: Patch

### CODE REVIEW

File: server/backends/invocationdb/invocationdb.go [hunk 1]
```diff
@@ -53,17 +53,27 @@ func (d *InvocationDB) createInvocation(tx *db.DB, ctx context.Context, ti *tabl
...
+	created := false
+	err := d.h.Transaction(ctx, func(tx *db.DB) error {
 		var existing tables.Invocation
 		if err := tx.Where("invocation_id = ?", ti.InvocationID).First(&existing).Error; err != nil {
 			if !db.IsRecordNotFound(err) {
 				return err
 			}
-			return d.createInvocation(tx, ctx, ti)
+			if err := d.createInvocation(tx, ctx, ti); err != nil {
+				created = true
```
Comment (tempoz) [inline server/backends/invocationdb/invocationdb.go:69]: The returned object from GORM's `Create` method contains [`RowsAffected`](https://gorm.io/docs/create.html#Create-Record), which would possibly be a better indicator for if we created a row than the value of the Error object is. Maybe change `createInvocation` to also return `RowsAffected`?

Comment (bduffany) [reply server/backends/invocationdb/invocationdb.go:69]: If the invocation already exists when calling createInvocation, then gorm should return an AlreadyExists error, so I think checking the error is OK

Comment (tempoz) [reply server/backends/invocationdb/invocationdb.go:69]: sounds good; just wanted to bring up the option.

**Review by tempoz — APPROVED** [review]
LGTM

**Pull request merged**